### PR TITLE
List component

### PR DESCRIPTION
**Pull request by mikitaaB**
https://github.com/RPE8/shiny-palm-tree/issues/21

### CODE REVIEW

File: src/components/initialDialog/content/content.tsx [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
 export const ContentDialog = (props: ContentDialogType) => {
 	return (
 		<div>
 			<p><a className="fa fa-home" href=""> MUI</a> / <a className="fa fa-fire" href="">Core</a> / <a className="fa fa-game" href="">Breadcrumb</a></p>
-			<ul className="listDialog">
-				<li>Single-line item</li>
-				<li>Single-line item</li>
-				<li>Single-line item</li>
-			</ul>
+			<List></List>
```
Comment (RPE8) [inline src/components/initialDialog/content/content.tsx:10]: How I can pass items into the List that need to be displayed?

Comment (mikitaaB) [reply src/components/initialDialog/content/content.tsx]: Fixed. By passing data to property 'items'

File: src/components/initialDialog/content/content.tsx [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
 export const ContentDialog = (props: ContentDialogType) => {
+	const listItems = [
+		{ "key": 0, "value": "First" },
+		{ "key": 1, "value": "Second" },
+		{ "key": 2, "value": "Third" }
+	];
 	return (
 		<div>
 			<p><a className="fa fa-home" href=""> MUI</a> / <a className="fa fa-fire" href="">Core</a> / <a className="fa fa-game" href="">Breadcrumb</a></p>
```
Comment (ExFroggy) [inline src/components/initialDialog/content/content.tsx:13]: Should i create another dialog component if i need its content to be different?

File: src/components/list/list.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import "./list.scss";
+
+interface ListType {
+	items: Array<{
+		key: number,
```
Comment (ExFroggy) [inline src/components/list/list.tsx:5]: Is it possible that the key will be a string type?

Comment (mikitaaB) [reply src/components/list/list.tsx]: Fixed

File: src/components/initialDialog/content/content.tsx (hunk 2 above)
Comment (mikitaaB) [reply src/components/initialDialog/content/content.tsx]: Fixed. Created a component 'Breadcrumbs'

**Comment by RPE8** [comment]
Is this pull request about List component only?
There are changes related to Breadcrumbs too

File: src/components/list/list.tsx (hunk 3 above)
Comment (mikitaaB) [reply src/components/list/list.tsx]: @ExFroggy

File: src/components/initialDialog/content/content.tsx (hunk 2 above)
Comment (mikitaaB) [reply src/components/initialDialog/content/content.tsx]: @ExFroggy Check the latest changes

**Comment by RPE8** [comment]
Why do you have changes related to Dialog here?)
I guess, there should be changes related to List component only, but not to Breadcrumbs and Dialog.

File: src/components/list/list.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+interface ListType {
+	items: Array<{
+		key: number | string,
+		value: string
+	}>
+}
+
+export const List = (props: ListType) => {
+	return (
+		<div className="list">
```
Comment (RPE8) [inline src/components/list/list.tsx:12]: BEM for css classes

File: src/components/list/list.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+			{props.items.map(item => (
+				<div key={item.key.toString()}><span className="listItem">{item.value}</span></div>
```
Comment (RPE8) [inline src/components/list/list.tsx:14]: BEM for css classes

File: src/components/list/list.tsx (hunk 4 above)
Comment (mikitaaB) [reply src/components/list/list.tsx:12]: Fixed

File: src/components/list/list.tsx (hunk 5 above)
Comment (mikitaaB) [reply src/components/list/list.tsx]: Fixed

File: src/components/dialog/dialog.tsx [hunk 6]
```diff
@@ -23,12 +23,8 @@ export default function Dialog(props: DialogType) {
 					}}
 					isOpen={props.isOpen}
 				>
-					<>
-						{props.header}
-					</>
-					<>
-						{props.content}
-					</>
+					{props.header}
```
Comment (RPE8) [inline src/components/dialog/dialog.tsx:26]: Why footer has a wrapper with specific class name, but content and header don't?

Comment (mikitaaB) [reply src/components/dialog/dialog.tsx:26]: To apply a className to a footer. Fixed

**Pull request merged**